app/src/test/java/com/concordium/wallet/data/util/NullableTypeAdapterFactoryShould.kt

```diff
@@ -0,0 +1,68 @@
+package com.concordium.wallet.data.util
+
+import com.concordium.wallet.core.gson.NullableTypeAdapterFactory
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+import com.google.gson.JsonParseException
+import org.junit.Assert
+import org.junit.Test
+
+/**
+ * Unit Test for the [NullableTypeAdapterFactory]
+ */
+class NullableTypeAdapterFactoryShould {
+
+    data class TestObject(
+        val type: String,
+        val sender: String,
+        val nullableValue: String?
+    )
+
+    private val gson: Gson = GsonBuilder()
+        .registerTypeAdapterFactory(NullableTypeAdapterFactory())
+        .create()
+
+    @Test
+    fun returnObjectWhenDataIsValid(){
+        val validJsonString ="{" +
+                "\"type\": \"test type\"," +
+                "\"sender\": \"test sender\"}"
+
+        val validObject = TestObject(type = "test type", sender = "test sender", nullableValue = null)
+
+        val testObject = gson.fromJson(validJsonString, TestObject::class.java)
+
+        Assert.assertEquals(
+            testObject.type,"test type"
+        )
+
+        Assert.assertEquals(
+            testObject.sender,"test sender"
+        )
+
+        Assert.assertEquals(
+            testObject, validObject
+        )
+    }
+
+    @Test
+    fun throwErrorWhenNonNullValueIsNotPresentInInput() {
```

Comment:
I would find it useful to have a case where the factory isn't registered to show how it would work without it. That would also test that it actually makes a difference to register.

Reply:
I will add another test.